numpyro/distributions/continuous.py

```diff
@@ -1188,6 +1191,169 @@ def infer_shapes(
+        between 0 (sites are independent, collapses to an iid multivariate normal) and
+        1 (perfect autocorrelation between sites), but the specification allows for negative
+        correlations.
+    :param float tau: positive precision for the multivariate normal
+    :param numpy.ndarray or scipy.sparse.csr_matrix W: symmetric adjacency matrix where 1
+        indicates adjacency between sites and 0 otherwise
+    :param bool is_sparse: whether to use a sparse form of W in calculations (must be True if
+        W is a :class:`scipy.sparse.spmatrix`)
+    """
+
+    arg_constraints = {
+        "loc": constraints.real_vector,
+        "alpha": constraints.open_interval(-1, 1),
+        "tau": constraints.positive,
+    }
+    support = constraints.real_vector
+    reparametrized_params = [
+        "loc",
+        "alpha",
+        "tau",
+    ]
+
+    def __init__(self, loc, alpha, tau, W, *, is_sparse=False, validate_args=None):
+        if jnp.ndim(loc) == 0:
+            (loc,) = promote_shapes(loc, shape=(1,))
+
+        self.is_sparse = is_sparse
+
+        # TODO: support batched W
+        if W.ndim != 2:
+            raise ValueError(
+                "Currently, we only support 2-dimensional W. Please make a feature request",
+                " if you need higher dimensional W.",
+            )
+        if not (isinstance(W, np.ndarray) or _is_sparse(W)):
+            raise ValueError(
+                "W needs to be a numpy array or a scipy sparse matrix. Please make a feature",
+                " request if you need to support jax ndarrays.",
+            )
+        if self.is_sparse:
+            # TODO: look into future jax sparse csr functionality and other developments
+            self.W = _to_sparse(W)
+        else:
+            assert not _is_sparse(
+                W
+            ), "W is a sparse matrix so please specify `is_sparse=True`."
+            # TODO: look into static jax ndarray representation
+            self.W = W
+
+        batch_shape = lax.broadcast_shapes(
+            jnp.shape(loc)[:-1],
+            jnp.shape(alpha),
+            jnp.shape(tau),
+            jnp.shape(self.W)[:-2],
+        )
+        event_shape = jnp.shape(self.W)[-1:]
+        (self.loc,) = promote_shapes(loc, shape=batch_shape + event_shape)
+        self.alpha, self.tau = promote_shapes(alpha, tau, shape=batch_shape)
+
+        super(CAR, self).__init__(
+            batch_shape=batch_shape,
+            event_shape=event_shape,
+            validate_args=validate_args,
+        )
+
+        if self._validate_args:
+            assert (
+                self.W.sum(axis=-1) > 0
+            ).all() > 0, "all sites in adjacency matrix W must have neighbours"
+
+            if self.is_sparse:
+                assert (self.W != self.W.T).nnz == 0, "W must be symmetric"
+            else:
+                assert np.array_equal(self.W, self.W.T), "W must be symmetric"
```

Comment:
`assert np.array_equal(self.W, np.swapaxes(self.W, -2, -1)), "W must be symmetric"`